js-src/compat/plugins/woo-carrier-agents/checkout-validation-woo-carrier-agents.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Manage checkout front-end validation for: WooCommerce Carrier Agents (by Markup.fi).
+ *
+ * DEPENDS ON:
+ * - checkout-validation.js // Main checkout validation script from Fluid Checkout
+ */
+
+(function (root, factory) {
+	if ( typeof define === 'function' && define.amd ) {
+		define([], factory(root));
+	} else if ( typeof exports === 'object' ) {
+		module.exports = factory(root);
+	} else {
+		root.CheckoutValidationWooCarrierAgents = factory(root);
+	}
+})(typeof global !== 'undefined' ? global : this.window || this.global, function (root) {
+
+	'use strict';
+
+	var _hasInitialized = false;
+	var _publicMethods = { };
+	var _settings = {
+		typeFieldSelector: '.validate-woo-carrier-agents',
+		sectionSelector: '.fc-shipping-method__packages',
+		validationMessages: {
+			pickup_point_not_selected: 'Selecting a pickup point is required before proceeding.',
+		},
+	};
+
+
+
+	/**
+	 * METHODS
+	 */
+
+
+
+	/**
+	 * Check if form row is a CNPJ field.
+	 * @param  {Field}    field            Field for validation.
+	 * @param  {Element}  formRow          Form row element.
+	 * @param  {String}   validationEvent  Event that triggered the validation.
+	 * @return {Boolean}                   Whether the field is a CNPF field.
+	 */
+	var isValidateField = function( field, formRow, validationEvent ) {
+		// Bail if not a woo carrier agents method field
+		if ( ! field.matches( _settings.typeFieldSelector ) ) { return false; }
+
+		return true;
+	};
+
+
+
+	/**
+	 * Validate if a Woo Carrier Agent shipping method collection point is selected.
+	 * @param  {Field}    field            Field for validation.
+	 * @param  {Element}  formRow          Form row element.
+	 * @param  {String}   validationEvent  Event that triggered the validation.
+	 * @return {Boolean}                   Whether a Woo Carrier Agent shipping method collection point has been selected.
+	 */
+	var validateField = function( field, formRow, validationEvent ) {
+		// Bail if Woo Carrier Agent shipping method field is empty
```

Comment:
- Avoid mentioning the related plugin on comments to make it easier to copy this file for another plugin if needed.